tests/platform_tests/api/test_fan.py

```diff
@@ -54,6 +62,28 @@ def setup(self, platform_api_conn):
             except:
                 pytest.fail("num_fans is not an integer")
 
+    #
+    # Helper functions
+    #
+
+    def compare_value_with_platform_facts(self, key, value, fan_idx):
+        expected_value = None
+
+        if self.chassis_facts:
+            expected_fan_drawers = self.chassis_facts.get("fan_drawers")
+            if expected_fan_drawers:
+                expected_fans = expected_fan_drawers[(fan_idx)/2].get("fans")
```

Comment:
This test should not be aware of fan drawers. It should test the fans which are directly on the chassis. There is no guarantee that all fan drawers will contain 2 fans.

It may be better if this test checked all fans on the platform. E.g., check all fans on the chassis, all fans in the fan drawers, all fans in PSUs.

Reply:
I think in that case, fan name should be tested inside fan drawers , psu's respectively , 
and we should not test the name here. 
or 
otherwise we need to gather all fans in a list and check their names. 
but this would be duplicate.

Reply:
For now, let's just test the fans which are directly on the chassis. We might want to make the fan testing code shared and utilitize it in every component which can contain fans.

Reply:
Please also update the fan_drawer test to check the fans in each drawer (for now, you can copy/paste code from here, we can break it out as shared code later).